src/vs/workbench/contrib/editSessions/browser/editSessionsStorageService.ts

```diff
@@ -196,6 +206,30 @@ export class EditSessionsWorkbenchService extends Disposable implements IEditSes
 		return authenticationSession !== undefined;
 	}
 
+	private cachedMachines: Map<string, string> | undefined;
+
+	async getMachineById(machineId: string) {
+		await this.initialize(false);
+
+		if (!this.cachedMachines) {
+			const machines = await this.machineClient!.getMachines();
```

Comment:
Maybe introduce a helper method `hasInitialized(): this is { machineClient: IUserDataSyncMachinesService; }` can help reduce the use of `!`.